platforms/nuttx/src/bootloader/nxp/imxrt_common/main.c

```diff
@@ -577,6 +582,15 @@ led_toggle(unsigned led)
 void
 arch_do_jump(const uint32_t *app_base)
 {
+	mpu_configure_region(IMXRT_ITCM_BASE, 256 * 1024,
```

Comment:
Maybe add a comment why we're doing this for example

Mark ITCM as RW in MPU so that 2nd stage boot can copy code into ITCM during startup.